torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2385,7 +2385,14 @@ def error_inputs_aminmax_amax_amin(op_info, device, **kwargs):
     min_values = torch.empty(L, dtype=torch.double, device=device)
     illegal_values = torch.empty(L, dtype=torch.int, device=device)
 
-    err_msg_amax_amin2 = "Expected the dtype for input and out to match"
+    if is_ref:
+        err_msg_amax_amin2 = ("Attempting to cast from torch.float32 to out "
+                              "tensor with dtype torch.int32, but this can't "
+                              "be cast because it is not safe!")
+    else:
+        err_msg_amax_amin2 = ("Attempting to cast from Float to out tensor "
+                              "with dtype Int, but this can't be cast because "
+                              "it is not safe!")
```

Comment:
I don't think we can unify error messages here because the CPP code will always use (or should use) CPP type names.